chain/manager.go

```diff
@@ -1086,47 +1086,86 @@ func (m *Manager) AddPoolTransactions(txns []types.Transaction) (known bool, err
 	return
 }
 
-// updateV2TransactionSet updates the basis of a transaction set to the current
-// tip. If the basis is already the tip, the transaction set is returned as-is.
-// Any transactions that were confirmed are removed from the set. Any ephemeral
-// state elements that were created by an update are updated.
+// UpdateStateElement updates the basis of a state element from "from" to "to".
+// If from and to are equal, the state element is not modified.
+func (m *Manager) UpdateStateElement(se *types.StateElement, from, to types.ChainIndex) error {
+	if from == to {
```

Comment:
needs a lock; which in turn means `AddV2PoolTransactions` needs to avoid double-locking

Reply:
Could you elaborate on why it needs a lock? I don't get it.

Reply:
I mean this method as a whole; we need to ensure a consistent view of the blockchain across this function